js/layout-javascript/agenda-code.js

```diff
@@ -1346,52 +1346,55 @@ DynamicList.prototype.renderDatesHTML = function(rows, index) {
   if (foundDateField) {
     dateField = foundDateField.column;
   }
-  // set first date in agenda
-  firstDate = new Date(clonedRecords[0].data[dateField]).toUTCString();
-
-  // set last date in agenda
-  lastDate = new Date(clonedRecords[clonedRecords.length - 1].data[dateField]).toUTCString();
-
-  // Adds 5 days before the first date
-  // Save them in an array
-  for (var i = 0; i < numberOfPlacholderDays; i++) { 
-    var newDate = {
-      week: moment(firstDate).utc().subtract(i + 1, 'days').format("ddd"),
-      day: moment(firstDate).utc().subtract(i + 1, 'days').format("DD"),
-      month: moment(firstDate).utc().subtract(i + 1, 'days').format("MMM"),
-      placeholder: true
+
+  if (clonedRecords.length) {
+    // set first date in agenda
+    firstDate = new Date(clonedRecords[0].data[dateField]).toUTCString();
```

Comment:
Is there a case where `clonedRecords[0]` will be `null?`

Reply:
I don't think so. A data source API returns an array, even when empty.